### PR TITLE
server: return already created ID for duplicated requests

### PR DESCRIPTION
**Pull request by haircommander**
turns out, kubelet does behave like this in some scenerios (static
pods). Instead of failing forever, return the already created pod.

Signed-off-by: Peter Hunt <pehunt@redhat.com>

#### What type of PR is this?

/kind bug
#### What this PR does / why we need it:
alternative to https://github.com/cri-o/cri-o/pull/6111

#### Does this PR introduce a user-facing change?

```release-note
Fix a bug where static pods cannot be created because they've already been created.
```

### CODE REVIEW

**Review by saschagrunert — APPROVED** [review]
/lgtm
/hold
Feel free to unhold if this is ready to be merged.

**Comment by TomSweeneyRedHat** [comment]
LGTM assuming happy tests

**Comment by haircommander** [comment]
for you, anything

**Review by saschagrunert — APPROVED** [review]
/lgtm

**Comment by haircommander** [comment]
/hold

I can't figure out why just kata is failing

@littlejawa do you have any insight?

**Review by TomSweeneyRedHat — COMMENTED** [review]
LGTM
once the tests are hapy

**Comment by littlejawa** [comment]
> /hold
> 
> I can't figure out why just kata is failing
> 
> @littlejawa do you have any insight?

Hey @haircommander,
I am not sure, but I think you're finding that this test (shm_size.bats : "check /dev/shm fails with incorrect values") actually never worked the way we thought with kata.
It is using the helpers.bat function `create_runtime_with_allowed_annotation`, and it seems we are skipping all the tests that use this function in kata because they fail - I suspect a bad interaction between this function, which changes the default runtime, and our test framework that makes sure kata is being used.
I am not sure that in the kata CI job, `create_runtime_with_allowed_annotation` actually changes the runtime being used by cri-o, and so the annotations are just not set.

The reason why we were not skipping this specific test is because it seemed to work, but looking at it, it is checking for failures, and doesn't verify the reason of the failure. So I am guessing that it was failing up to now because annotations were just not taken into account, and this made the test "pass", for the wrong reasons.

I am not sure why it changed its behavior now, but apparently your change makes the second call to `crictl runp` find an existing pod and returning it (this is what the cri-o log show), ignoring the potential issue with annotation, and so reporting success?
I don't know why the pod is still around at that time though.

Anyway - TL;DR: I think you should ignore this error, and I will modify our test script to skip this test, until we fix the function in question.

**Comment by haircommander** [comment]
/hold

actually, ci is broken at the moment, let's wait till it's ready

**Comment by saschagrunert** [comment]
@haircommander can you please rebase that we can merge it soon'ish?

**Review by saschagrunert — APPROVED** [review]

**Pull request merged**
